### DISCUSSION TITLE
gitlab with MFA authentication

### DISCUSSION THREAD

**Discussion by rlmfc** [keilerkonzept/terraform-module-versions#61] [Q&A]
Getting error on check with all MFA enabled gitlab repositories - 
"[terraform-module-versions] error: fetch versions from <reponame>: git list refs: authentication required"

Tried setting SSH insteadOf HTTPS, and using git credential-helper, but nothing seems to work.

Any ideas ?

**Comment by sgreben** [comment]
Hey @rlmfc ! I think with 2FA enabled you need to use <https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html> or <https://docs.gitlab.com/ee/security/two_factor_authentication.html#2fa-for-git-over-ssh-operations>.

For the access token route, we could simply add a generic Basic Auth option for git, analogously to https://github.com/keilerkonzept/terraform-module-versions/blob/9e8c803363a2f84002ad84d4a2b2263aee1d16a0/main.go#L150-L154

I can take a stab at it sometime this week (depending on work workload..).
